Assets/MixedRealityToolkit/Graph/GraphConnector.cs

```diff
@@ -0,0 +1,198 @@
+        /// </summary>
+        /// <returns>The Unity coroutine.</returns>
+        public IEnumerator SignIn()
+        {
+            var authTask = SignInAsync();
+            while (!authTask.IsCompleted)
+            {
+                yield return null;
+            }
+        }
+
+        /// <summary>
+        /// Get an access token for the graph scopes and app id. An attempt is first made to 
+        /// acquire the token silently. If that fails, then we try to acquire the token by prompting the user.
+        /// </summary>
+        /// <returns>The async task.</returns>
+        public Task SignInAsync()
+        {
+            return graphAuthentication.SignInAsync(graphAccessScopes);
+        }
+
+        /// <summary>
+        /// Method to sign out the current user.
+        /// </summary>
+        public void SignOut()
+        {
+            graphAuthentication.SignOut();
+        }
+
+        /// <summary>
+        /// Makes GET request via Graph's REST APIs.
+        /// </summary>
+        /// <typeparam name="T">The returned type for graph response.</typeparam>
+        /// <param name="graphRequest">The graph request to make.</param>
+        /// <returns>The Unity coroutine.</returns>
+        public IEnumerator MakeRequestGet<T>(GraphRequest<T> graphRequest) where T : GraphResponse, new()
+        {
+            using (UnityWebRequest webRequest = UnityWebRequest.Get(graphRequest.Uri))
+            {
+                yield return MakeRequest(graphRequest, webRequest);
+            }
+        }
+
+        /// <summary>
+        /// Makes GET request via Graph's REST APIs. 
+        /// </summary>
+        /// <typeparam name="T">The returned type for graph response.</typeparam>
+        /// <param name="endpoint">The Graph endpoint to call.</param>
+        /// <param name="api">The Graph API to call.</param>
+        /// <returns>The async task.</returns>
+        public Task<T> MakeRequestGetAsync<T>(string endpoint, string api) where T : GraphResponse, new()
+        {
+            GraphRequest<T> graphRequest = new GraphRequest<T>(endpoint, api);
+
+            StartCoroutine(MakeRequestGet<T>(graphRequest));
+
+            return Task.Run(() =>
+            {
+                graphRequest.WaitForCompletion();
+
+                var result = graphRequest.Result;
+                graphRequest.Dispose();
+
+                return result;
+            });
+        }
+
+        /// <summary>
+        /// Makes POST request via Graph's REST APIs.
+        /// </summary>
+        /// <typeparam name="T">The returned type for graph response.</typeparam>
+        /// <param name="graphRequest">The graph request to make.</param>
+        /// <returns>The Unity coroutine.</returns>
+        public IEnumerator MakeRequestPost<T>(GraphRequest<T> graphRequest) where T : GraphResponse, new()
+        {
+            using (UnityWebRequest webRequest = UnityWebRequest.Post(graphRequest.Uri, graphRequest.Body))
```

Comment:
Let's use the REST library found in MixedRealityToolkit/_Core/Utilities/WebRequestRest

Reply:
Oh, I like it.